proxy/src/config.rs

```diff
@@ -0,0 +1,35 @@
+use serde::Deserialize;
+use std::error::Error;
+use std::fs::File;
+
+#[derive(Debug, Deserialize)]
+pub struct Config {
+    pub routes: Vec<Route>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Route {
+    #[serde(rename = "match")]
+    pub match_criteria: Match,
+    #[serde(rename = "route")]
+    pub route_action: RouteAction,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Match {
+    pub host: String,
+    pub path_prefix_pattern: String,
```

Comment:
I think we should make this an Option<String>

It would allow matching on the host name only, and not matching the path. For single cell in a region that would be fine.